logstash-core/lib/logstash/webserver.rb

```diff
@@ -8,19 +8,20 @@
 module LogStash 
   class WebServer
 
+    class HostBindingError < StandardError; end
+
     extend Forwardable
 
     attr_reader :logger, :status, :config, :options, :cli_options, :runner, :binder, :events
 
     def_delegator :@runner, :stats
 
-    DEFAULT_HOST = "127.0.0.1".freeze
```

Comment:
I can implement it like this, no problem. however it looks like too much brute force as this will also check not just when logstash runs, but also if they have other devices using one port in our range (and this we never know). I'm sure we will never use the full set, but a few might happen. 

with acking puma to boot to check this we might prevent a possible race condition, but will have a slowdown. 

I'm fine with it.